src/thindev.rs

```
@@ -78,6 +88,28 @@ impl ThinDev {
 
     }
 
+    /// Get the current status of the thin device.
+    pub fn status(&self, dm: &DM) -> DmResult<ThinStatus> {
+        let (_, mut status) = try!(dm.table_status(&DevId::Name(&self.dev_info.name()),
+                                                   DmFlags::empty()));
+
+        if status.len() != 1 {
+            return Err(DmError::Io(io::Error::new(io::ErrorKind::InvalidData,
+                                                  "Expected 1 line from thin status")));
+        }
+
+        // We should either get 1 line or the kernel is broken
+        let status_line = status.pop().expect("status didn't have one line").3;
+        if status_line.starts_with("Fail") {
+            return Ok(ThinStatus::Fail);
+        }
+        let status_vals = status_line.split(' ').collect::<Vec<_>>();
+
```

Properly speaking, you should return an error if ```status_vals``` has length 0 as in PR #20.